esrally/rally.py

```diff
@@ -171,6 +171,135 @@ def positive_number(v):
+        default=""
+    )
+    install_parser.add_argument(
+        "--elasticsearch-plugins",
+        help="Define the Elasticsearch plugins to install. (default: install no plugins).",
+        default="")
+    install_parser.add_argument(
+        "--plugin-params",
+        help="Define a comma-separated list of key:value pairs that are injected verbatim to all plugins as variables.",
+        default=""
+    )
+    install_parser.add_argument(
+        "--network-host",
+        help="The IP address to bind to and publish",
+        default="127.0.0.1"
+    )
+    install_parser.add_argument(
+        "--http-port",
+        help="The port to expose for HTTP traffic",
+        default="39200"
+    )
+    install_parser.add_argument(
+        "--node-name",
+        help="The name of this Elasticsearch node",
+        default="rally-node-0"
+    )
+    install_parser.add_argument(
+        "--master-nodes",
+        help="A comma-separated list of the initial master node names",
+        default=""
+    )
+    install_parser.add_argument(
+        "--seed-hosts",
+        help="A comma-separated list of the initial seed host IPs",
+        default=""
+    )
+
+    start_parser = subparsers.add_parser("start", help="Starts an Elasticsearch node locally")
+    start_parser.add_argument(
+        "--installation-id",
+        required=True,
+        help="The id of the installation to start",
+        # the default will be dynamically derived by racecontrol based on the presence / absence of other command line options
+        default="")
+    start_parser.add_argument(
+        "--race-id",
+        required=True,
+        help="Define a unique id for this race.",
+        default="")
+    start_parser.add_argument(
+        "--runtime-jdk",
+        type=positive_number,
+        help="The major version of the runtime JDK to use.",
+        default=None)
+    start_parser.add_argument(
+        "--telemetry",
+        help="Enable the provided telemetry devices, provided as a comma-separated list. List possible telemetry devices "
+             "with `%s list telemetry`." % PROGRAM_NAME,
+        default="")
+    start_parser.add_argument(
+        "--telemetry-params",
+        help="Define a comma-separated list of key:value pairs that are injected verbatim to the telemetry devices as parameters.",
+        default=""
+    )
+
+    stop_parser = subparsers.add_parser("stop", help="Stops an Elasticsearch node locally")
+    stop_parser.add_argument(
+        "--installation-id",
+        required=True,
+        help="The id of the installation to start",
+        # the default will be dynamically derived by racecontrol based on the presence / absence of other command line options
+        default="")
+    stop_parser.add_argument(
+        "--race-id",
+        required=True,
+        help="Define a unique id for this race.",
+        default="")
```

Comment:
Presumably, the "stop" command shouldn't be caring about races as it is just stopping nodes?

Reply:
Upon stopping a node, Rally needs to store metrics and aggregate them. To have consistent meta-data across all nodes involved in a benchmark, we need the race id to tie them together and that's why this command line parameter is needed. We could probably store it when a node is started but I've opted for this simpler approach here. In fact, we could even skip it upon node startup because we only store metrics when a node is stopped but this felt like leaking implementation details into the command line interface which I wanted to avoid.

Reply:
I like the approach of storing the active race ID.  Any chance we could include it in this PR?

Reply:
Good idea, it makes the interface definitely simpler and easier to use. I've implemented this in https://github.com/elastic/rally/pull/830/commits/dbbbdb40e4d3fe8c172c75bc626256d13205d210.